subworkflows/local/analyse_transcripts.nf

```diff
@@ -11,22 +11,22 @@ include { DROP_FILTER_RESULTS } from '../../modules/local/drop_filter_results'
 
 workflow ANALYSE_TRANSCRIPTS {
     take:
-        ch_bam_bai                    // channel (mandatory): [ val(meta), [ path(bam) ],[ path(bai) ] ]
-        ch_bam_ds_bai                 // channel (mandatory): [ val(meta), [ path(bam) ],[ path(bai) ] ]
-        ch_gtf                        // channel (mandatory): [ path(gtf) ]
-        ch_fasta_fai_meta             // channel (mandatory): [ val(meta), [ path(fasta), path(fai) ]
-        gene_counts                   // channel [val(meta), path(tsv)]
-        ch_ref_drop_count_file        // channel [ path(tsv) ]
-        ch_ref_drop_annot_file        // channel [ path(tsv) ]
-        ch_ref_drop_splice_folder     // channel [ path(folder) ]
-        genome                        // channel [val(genome)]
-        drop_group_samples_ae         // channel [val(drop_group_samples_ae)]
-        drop_group_samples_as         // channel [val(drop_group_samples_as)]
-        drop_padjcutoff_ae            // channel [val(drop_padjcutoff_ae)]
-        drop_padjcutoff_as            // channel [val(drop_padjcutoff_as)]
-        drop_zscorecutoff             // channel [val(drop_zscorecutoff)]
-        ch_gene_panel_clinical_filter // channel [ path(tsv) ]
-        case_info                     // channel [val(case_id)]
+        ch_bam_bai                    // channel [mandatory]: [ val(meta), [ path(bam) ],[ path(bai) ] ]
+        ch_bam_ds_bai                 // channel [mandatory]: [ val(meta), [ path(bam) ],[ path(bai) ] ]
+        ch_gtf                        // channel [mandatory]: [ val(meta), [ path(gtf) ] ]
+        ch_fasta_fai                  // channel [mandatory]: [ val(meta), [ path(fasta), path(fai) ]
+        gene_counts                   // channel: [ val(meta), path(tsv) ]
+        ch_ref_drop_count_file        // channel: [ path(tsv) ]
+        ch_ref_drop_annot_file        // channel: [ path(tsv) ]
+        ch_ref_drop_splice_folder     // channel: [ path(folder) ]
+        genome                        // parameter: [ val(genome) ]
+        drop_group_samples_ae         // parameter: [ val(drop_group_samples_ae) ]
+        drop_group_samples_as         // parameter: [ val(drop_group_samples_as) ]
+        drop_padjcutoff_ae            // parameter: [ val(drop_padjcutoff_ae) ]
+        drop_padjcutoff_as            // parameter: [ val(drop_padjcutoff_as) ]
+        drop_zscorecutoff             // parameter: [ val(drop_zscorecutoff) ]
```

Comment:
Consider updating these with the type they actually are as well, you can check with `println genome.getClass()`.

Reply:
Will do, it does make it more clear

Reply:
I have doubts about this option. One major advantage of nextflow is that knowledge of java/groovy is not required. So, I believe it is not ideal to use class names from groovy to define what's being passed around. Simply indicating whether something is a channel or a parameter should be enough in my view. There is no need for users to know whether something is an array/list/arrayList, and it could add to confusion. 😅

Reply:
Yeah that''s totally fair, what do you think about `string`, `boolean`, `integer`, `float`?

Reply:
Same as array/list/arrayList actually, because we don't really interact with objects at the level in the workflow. Types of parameters are defined in the nextflow_schema.json, and once they enter the workflow, they are handled as parameters.

Reply:
Alright!